.github/workflows/publish.yml

```diff
@@ -0,0 +1,67 @@
+on:
+  workflow_call:
+    inputs:
+      file_path:
+        required: true
+        type: string
+      release_name:
+        required: true
+        type: string
+      tag:
+        required: true
+        type: string
+      changelog:
+        required: true
+        type: string
+      cf_mc_version:
+        required: true
+        type: string
+      mr_mc_version:
+        required: true
+        type: string
+      mr_loader:
+        required: true
+        type: string
+    secrets: inherit
+
+jobs:
+  release-to-cf:
+    name: Release to CurseForge
+    runs-on: ubuntu-latest
+    needs: [changelog, artifact]
+    steps:
+      - uses: actions/download-artifact@v3
+
+      - name: Upload to CurseForge
+        uses: itsmeow/curseforge-upload@v3
+        with:
+          file_path: "${{ inputs.file_path }}"
+          game_endpoint: minecraft
+          game_versions: "${{ inputs.cf_mc_version }}"
+          project_id: 303451
+          token: ${{ secrets.CF_API_TOKEN }}
+          release_type: alpha
+          display_name: "${{ inputs.release_name }}"
+          changelog_type: markdown
+          changelog: "${{ needs.changelog.outputs.changelog }}"
+
+  release-to-modrinth:
+    name: Release to Modrinth
+    runs-on: ubuntu-latest
+    needs: [changelog, artifact]
+    steps:
+      - uses: actions/download-artifact@v3
+
+      - name: Upload to Modrinth
+        uses: RubixDev/modrinth-upload@v1.0.0
+        with:
+          token: ${{ secrets.MODRINTH_API_TOKEN }}
+          file_path: "${{ inputs.file_path }}"
```

Comment:
We may be able to move Modrinth a as it supports multiple files per version, but i couldn’t find whether the action supported it.